From 6d2944b38daa240f24df63ccbeb2f75470299cd9 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Sun, 04 Aug 2024 18:34:40 +0300
Subject: [PATCH] [FSW-7946] fs1: ppboot: onix: remove early-write-resp

This patch is removing early write response from
fs1 onix, because it is not supported in fs1 Onix.
This mechanism is used for performance enhancement,
hence onix validation can be done even without this
feature.

Change-Id: Id14172978ca1acbc9dda449b240c78944a57e765
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 338213e..84f3c42 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -207,7 +207,7 @@
 				 1 << ARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_AWCACHE_OVRD_EN_S |\
 				 1 << ARC_LOCAL_CONF_ARC_LBU_AXI_SPLIT_CTRL2_ARCACHE_OVRD_EN_S)
 
-static void hl_lbw_early_write_resp_enable(void)
+__weak void hl_lbw_early_write_resp_enable(void)
 {
 	u32 axi_split_val, val;
 
diff --git a/arch/fs1/arc/onix.c b/arch/fs1/arc/onix.c
index 954f7be..82ca3cc 100644
--- a/arch/fs1/arc/onix.c
+++ b/arch/fs1/arc/onix.c
@@ -57,3 +57,6 @@
 
 /* VUART not available in Onix */
 void preboot_putc_virt_uart(const char c) {}
+
+/* Early write response not available in Onix */
+void hl_lbw_early_write_resp_enable(void) {}
